test/lit/passes/cfp-copies.wast

```diff
@@ -0,0 +1,8064 @@
+    (struct.get $B 0
+      (local.get $B)
+    )
+  )
+)
+
+(module
+  (rec
+    ;; CHECK:      (rec
+    ;; CHECK-NEXT:  (type $super (sub (struct (field (mut i32)))))
+    (type $super (sub (struct (field (mut i32)))))
+    ;; CHECK:       (type $struct (sub $super (struct (field (mut i32)))))
+    (type $struct (sub $super (struct (field (mut i32)))))
+    ;; CHECK:       (type $sub (sub $struct (struct (field (mut i32)))))
+    (type $sub (sub $struct (struct (field (mut i32)))))
+    ;; CHECK:       (type $other (struct (field (mut i32))))
+    (type $other (struct (field (mut i32))))
+  )
+
+  ;; CHECK:      (type $4 (func))
+
+  ;; CHECK:      (type $5 (func (param (ref $other) (ref $struct))))
+
+  ;; CHECK:      (func $init (type $4)
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (struct.new $other
+  ;; CHECK-NEXT:    (i32.const 10)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $init
+    (drop
+      (struct.new $other
+        (i32.const 10)
+      )
+    )
+  )
+
+  ;; CHECK:      (func $copy (type $5) (param $other (ref $other)) (param $struct (ref $struct))
+  ;; CHECK-NEXT:  (struct.set $struct 0
+  ;; CHECK-NEXT:   (local.get $struct)
+  ;; CHECK-NEXT:   (block (result i32)
+  ;; CHECK-NEXT:    (drop
+  ;; CHECK-NEXT:     (ref.as_non_null
+  ;; CHECK-NEXT:      (local.get $other)
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (i32.const 10)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $copy (param $other (ref $other)) (param $struct (ref $struct))
+    ;; Copy from $other to $struct.
+    (struct.set $struct 0
+      (local.get $struct)
+      (struct.get $other 0
+        (local.get $other)
+      )
+    )
+  )
+
+  ;; CHECK:      (func $gets (type $4)
+  ;; CHECK-NEXT:  (local $super (ref null $super))
+  ;; CHECK-NEXT:  (local $struct (ref null $struct))
+  ;; CHECK-NEXT:  (local $sub (ref null $sub))
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (block (result i32)
+  ;; CHECK-NEXT:    (drop
+  ;; CHECK-NEXT:     (ref.as_non_null
+  ;; CHECK-NEXT:      (local.get $super)
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (i32.const 10)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (block (result i32)
+  ;; CHECK-NEXT:    (drop
+  ;; CHECK-NEXT:     (ref.as_non_null
+  ;; CHECK-NEXT:      (local.get $struct)
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (i32.const 10)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (block (result i32)
+  ;; CHECK-NEXT:    (drop
+  ;; CHECK-NEXT:     (ref.as_non_null
+  ;; CHECK-NEXT:      (local.get $sub)
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (i32.const 10)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $gets
+    (local $super (ref null $super))
+    (local $struct (ref null $struct))
+    (local $sub (ref null $sub))
+    (drop
```

Comment:
Please add a comment here. I think it can say "we never wrote to $super, so the only things we can read are what was written to $struct (or $sub, if the reference in $copy was to a subtype)"